src/test/ui/wf/issue-87495.rs

```diff
@@ -0,0 +1,11 @@
+// Regression test for the ICE described in #87495.
+
+#![allow(non_camel_case_types,bare_trait_objects)]
+
+trait u32 {
+    const CONST: u32;
```

Comment:
Can you change this to `const CONST: (bool, dyn TraitName);` ? This will verify that we're properly performing the HIR-based WF check (we should get a span pointing at the `dyn TraitName`).

Reply:
Fixed.